packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -57,6 +58,164 @@ function findInputObjectType(input: IntrospectionInputValue, schema: Introspecti
     return filterType;
 }
 
+function getRequiredQueryArgs({ gridQueryType }: { gridQueryType: IntrospectionField }) {
+    const skipParams = ["offset", "limit", "sort", "search", "filter", "scope"];
+    const requiredParams: IntrospectionInputValue[] = [];
+    gridQueryType.args.forEach((arg) => {
+        if (skipParams.includes(arg.name)) return;
+        if (arg.type.kind === "NON_NULL") {
+            requiredParams.push(arg);
+        }
+    });
+    return requiredParams;
+}
+
+function getRequiredMutationArgs({ createMutationType }: { createMutationType: IntrospectionField }) {
+    const skipParams = ["input"];
+    const requiredParams: IntrospectionInputValue[] = [];
+    createMutationType.args.forEach((arg) => {
+        if (skipParams.includes(arg.name)) return;
+        if (arg.type.kind === "NON_NULL") {
+            requiredParams.push(arg);
+        }
+    });
+    return requiredParams;
+}
+
+function getRequiredGqlArgs({
+    config,
+    gridQueryType,
+    createMutationType,
+    gqlIntrospection,
+}: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    config: GridConfig<any>;
+    gridQueryType: IntrospectionField;
+    createMutationType?: IntrospectionField;
+    gqlIntrospection: IntrospectionQuery;
+}) {
+    const params: IntrospectionInputValue[] = [];
+    getRequiredQueryArgs({ gridQueryType }).forEach((param) => {
+        params.push(param);
+    });
+    if (createMutationType) {
+        getRequiredMutationArgs({ createMutationType }).forEach((arg) => {
+            // skip similar params
+            if (params.find((addedArg) => addedArg.name === arg.name)) return;
+            params.push(arg);
+        });
+    }
+
+    const requiredParams: { type: string; name: string }[] = [];
+    params.forEach((arg) => {
+        if (arg.type.kind !== "NON_NULL") return;
```

Comment:
You're only pushing NON_NULL params into this array, so you may remove it.

Reply:
I need this line because typescript does not infer the type correctly